internal/provider/organizations_inventory_device_datasource_test.go

```diff
@@ -0,0 +1,113 @@
+				),
+			},
+
+			// TODO: Create and Read OrganizationsInventoryDevices
+			//{
+			//	Config: testAccOrganizationsInventoryDevicesDataSourceConfigCreate,
+			//	Check: resource.ComposeAggregateTestCheckFunc(
+			//		resource.TestCheckResourceAttr("organizations_inventory_devices.test", "id", "example-id"),
+			//	),
+			//},
+
+			// Read OrganizationsInventoryDevices
+			{
+				Config: testAccOrganizationsInventoryDeviceDataSourceConfigRead,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.meraki_organizations_inventory_device.test", "id", "example-id"),
+				),
+			},
+		},
+	})
+}
+
+// testAccOrganizationsInventoryDeviceDataSourceConfigCreateOrganization is a constant string that defines the configuration for creating an organization resource in your tests.
+const testAccOrganizationsInventoryDeviceDataSourceConfigCreateOrganization = `
+ resource "meraki_organization" "test" {
+ 	name = "test_meraki_organizations_inventory_device"
+ 	api_enabled = true
+ }
+ `
+
+// testAccOrganizationsInventoryDeviceDataSourceConfigCreateNetwork is a constant string that defines the configuration for creating a network resource in your tests.
+// It depends on the organization resource.
+const testAccOrganizationsInventoryDeviceDataSourceConfigCreateNetwork = `
+resource "meraki_organization" "test" {}
+
+resource "meraki_network" "test" {
+	depends_on = [resource.meraki_organization.test]
+	organization_id = resource.meraki_organization.test.organization_id
+	product_types = ["appliance", "switch", "wireless"]
+	tags = ["tag1"]
+	name = "Main Office"
+	timezone = "America/Los_Angeles"
+	notes = "Additional description of the network"
+}
+`
+
+// TODO: Make a change to the configuration to test
+// testAccOrganizationsInventoryDevicesDataSourceConfigCreate is a constant string that defines the configuration for creating and updating a organizations_{organizationId}_inventory_devices resource in your tests.
+// It depends on both the organization and network resources.
+//const testAccOrganizationsInventoryDevicesDataSourceConfigCreate = `
+//resource "meraki_organization" "test" {}
+//resource "meraki_network" "test" {
+//	depends_on = [resource.meraki_organization.test]
+//	product_types = ["appliance", "switch", "wireless"]
+//}
+//
+//resource "meraki_organizations_inventory_device" "test" {
+//	depends_on = [resource.meraki_network.test, resource.meraki_organization.test]
+//  	network_id = resource.meraki_network.test.network_id
+//}
+//`
+
+// testAccOrganizationsInventoryDeviceDataSourceConfigRead is a constant string that defines the configuration for creating and updating a organizations_{organizationId}_inventory_devices resource in your tests.
+// It depends on both the organization and network resources.
+const testAccOrganizationsInventoryDeviceDataSourceConfigRead = `
+
+data "meraki_organizations_inventory_device" "test" {
+  	organization_id = "1234641"
```

Comment:
instead of hard coding we can use env variable TF_ACC_MERAKI_ORGANZIATION_ID, TF_ACC_MERAKI_MX_SERIA and pass it via function logic example : https://github.com/core-infra-svcs/terraform-provider-meraki/blob/main/internal/provider/organizations_licenses_datasource_test.go